os/pm/pm_idle.c

```diff
@@ -65,22 +65,54 @@ void pm_idle(void)
 	enum pm_state_e newstate;
 #ifdef CONFIG_PM_TIMEDWAKEUP
 	clock_t delay;
+#endif
+#ifdef CONFIG_SMP
+	int cpu;
+	int paused_count = 0;
+	FAR struct tcb_s *tcb = NULL;
 #endif
 	/* Decide, which power saving level can be obtained */
 	flags = enter_critical_section();
 	newstate = pm_checkstate();
 	/* Perform state-dependent logic here */
-	pmvdbg("newstate= %d\n", newstate);
+	/* For SMP case, we need to check secondary core status
+	   If secondary core status is not in idle thread, abort
+	   the sleep and check again on next cycle
+	*/
+#ifdef CONFIG_SMP
+	if (newstate == PM_SLEEP) {
+		for (cpu = 1; cpu < CONFIG_SMP_NCPUS; cpu++) {
+			tcb = current_task(cpu);
+			if (tcb->pid != cpu) {
+				paused_count = cpu - 1;
+				goto EXIT;
+			}
+			(void)sched_pause_cpu(tcb);
+		}
+	}
+#endif
 	/* Then force the global state change */
 	if (pm_changestate(newstate) < 0) {
 		/* The new state change failed */
 		pmdbg("State change failed! newstate = %d\n", newstate);
 		goto EXIT;
+	} else {
+		pmvdbg("newstate= %d\n", newstate);
 	}
 	/* If current state is not good to go sleep then do core power saving*/
 	if (g_pmglobals.state != PM_SLEEP) {
 		goto EXIT;
 	}
+#ifdef CONFIG_SMP
+	/* Send signal to shutdown other cores here */
+	for (cpu = 1; cpu < CONFIG_SMP_NCPUS; cpu++) {
+		up_cpu_hotplug(cpu);
```

Comment:
If we perform the hotplug operation here, then better remove duplicate SGI4 from up_pm_board_sleep().

Reply:
Updated~